### ISSUE TITLE
Flake8 (Hound CI) rule E402 breaks AppVeyor build

### ISSUE THREAD

**Issue by JornVernee** [mypaint/mypaint#919]
This is an issue for the problem discussed in this community forum thread (as requested):

https://community.mypaint.org/t/flake8-hound-ci-rule-e402-breaks-appveyor-build/1178

**Comment by achadwick** [comment]
Thanks for reporting, all!  We should not be disabling the rule globally, but yes: the import order stuff is quite finicky.

If you need to disable a rule on a specific line, please append `# noci: E402` or similar to the line(s) you want to be excused for: there are good reasons for doing things out of order when it comes to `gi.require_version()` stuff.

The actual error from the [build log for your branch](https://ci.appveyor.com/project/achadwick/mypaint/build/1.3.0-alpha+master.291/job/mrhbi72ocfpl06nt) is 

```
======================================================================
ERROR: setUpClass (tests.rendering.Scroll)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "c:/projects/mypaint/tests/rendering.py", line 317, in setUpClass
    tdw = TiledDrawWidget()
  File "c:/projects/mypaint/tests/rendering.py", line 308, in __init__
    .__init__(self, *args, **kwargs)
  File "c:/projects/mypaint/gui/tileddrawwidget.py", line 133, in __init__
    from . import application
  File "c:/projects/mypaint/gui/application.py", line 61, in <module>
    from . import drawwindow  # noqa: F401
  File "c:/projects/mypaint/gui/drawwindow.py", line 45, in <module>
    import gui.footer
  File "c:/projects/mypaint/gui/footer.py", line 16
    from __future__ import division, print_function
SyntaxError: from __future__ imports must occur at the beginning of the file
----------------------------------------------------------------------
```

Which isn't the linter: that's Python itself! The commit https://github.com/JornVernee/mypaint/commit/7f0815289b34411396630f15921fa2a7663e0f5c introduced something that was syntactically invalid: specifically, all `__future__` imports *MUST* be first in the file for the file itself to be valid Python.

I will introduce some fixes for this module and my current `flake8` onto master and test it just so we can exclude any possible interaction between Appveyor and code that keeps Hound-CI (and flake8) happy. There should be none: the two integrations have different jobs and work differently :)

**Comment by JornVernee** [comment]
> all `__future__` imports _MUST_ be first in the file for the file itself to be valid Python.

Good to know! It didn't break the travis and tea checks though

**Comment by achadwick** [comment]
For what it's worth, I favour the following block order for imports, but it's not strictly enforced anywhere. A single blank line between each block to keep things split.

1. Python's "batteries included" standard modules e.g. `json`, `logging`, or `sys`. `__future__` goes first.
2. Third-party modules like `from gi.repository import ...` or `import cairo`.
3. MyPaint's own module library, e.g. `from lib.pycompat import ...`.

No reason other than aesthetics really, and the order sometimes switches around as you'll have noticed. Definition of the de facto standard "logger" variable for the current module tends to drift to the end of these blocks.

**Comment by JornVernee** [comment]
Whoops

**Comment by achadwick** [comment]
It's cool :)

**Comment by achadwick** [comment]
It seems that this is still not a bug. I've pushed 8f08d5e06edb6bc452c517216478dbd1b6e450c6 to a testing branch to make sure that fixing the one module solidly identified so far for flake8 lint reasons doesn't break the builds on AppVeyor or Tea-CI.

Sometimes there are metaprogramming magic reasons for particular import orders. I want to work towards labelling those, because they can be a pain.

> It [not having __future__ first?] didn't break the travis and tea checks though

Er... yes it did unless I'm I'm reading something?

**Comment by JornVernee** [comment]
Idk, I was updating my PR, when Hound CI wanted me to put the module level imports at the top of the file. So I did that, putting them before the `__future__` import (https://github.com/mypaint/mypaint/pull/917/commits/7f0815289b34411396630f15921fa2a7663e0f5c). Locally, everything still built find (iirc), but on github only the appveyor check failed.

Previously it was also only appveyor that failed when I a non-local (not in a function) import of `application`.

But I was rapidly pushing commits, so some of the checks might have been running for some of the older pushes I did? (that seemed to happen at least once with Hound, that's when I made this: https://community.mypaint.org/t/see-which-violations-hound-ci-is-failing-on/1176).
